src/NServiceBus.Core/Pipeline/BehaviorContext.cs

```diff
@@ -8,16 +9,21 @@
     /// </summary>
     public abstract class BehaviorContext : ContextBag
     {
+        //TODO DanielTim: Does it make to force us to provide a RootContext here (to ensure Settings are set)? RootContext currently is private.
         /// <summary>
         /// Create an instance of <see cref="BehaviorContext"/>.
         /// </summary>
         /// <param name="parentContext">The parent context.</param>
-        protected BehaviorContext(ContextBag parentContext) : base(parentContext)
+        protected BehaviorContext(BehaviorContext parentContext) : base(parentContext)
         {
-          
         }
 
         /// <summary>
+        /// The readonly settings.
+        /// </summary>
+        public ReadOnlySettings Settings => Get<ReadOnlySettings>();
```

Comment:
I've always seen the settings as "config time only" and using setting at "runtime" is a smell IMO. Why do we need to make them available at runtime? (I know that we currently provide the settings via DI :(  )